File: app/src/main/java/com/example/sena/Registro.java
package com.example.sena;

import android.content.Intent;
import android.os.Bundle;
import android.util.Patterns;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.button.MaterialButton;

public class Registro extends AppCompatActivity {

    private EditText etCorreo, etUsuario, etPass;
    private MaterialButton btnCrearCuenta, btnRegresar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registro);

        etCorreo = findViewById(R.id.etCorreo);
        etUsuario = findViewById(R.id.etUsuario);
        etPass = findViewById(R.id.etContrasenaReg);

        btnCrearCuenta = findViewById(R.id.btnCrearCuenta);
        btnRegresar = findViewById(R.id.btnRegresar);

        btnRegresar.setOnClickListener(v -> {
            startActivity(new Intent(Registro.this, MainActivity.class));
            finish();
        });

        btnCrearCuenta.setOnClickListener(v -> {
            String correo = etCorreo.getText().toString().trim();
            String user = etUsuario.getText().toString().trim();
            String pass = etPass.getText().toString();

            if (!Patterns.EMAIL_ADDRESS.matcher(correo).matches()) {
                Toast.makeText(this, "Correo inválido.", Toast.LENGTH_SHORT).show();
                return;
            }

            int res = Administracion.register(this, correo, user, pass);

            if (res == 1) {
                Toast.makeText(this, "Completa todos los campos.", Toast.LENGTH_SHORT).show();
                return;
            }
            if (res == 2) {
                Toast.makeText(this, "Ese usuario ya existe.", Toast.LENGTH_SHORT).show();
                return;
            }

            Toast.makeText(this, "Cuenta creada. Bienvenido " + user, Toast.LENGTH_SHORT).show();
            startActivity(new Intent(Registro.this, Inicio.class));
            finish();
        });
    }
}

File: app/src/main/java/com/example/sena/Registro.kt
package com.example.sena

import android.content.Intent
import android.os.Bundle
import android.widget.EditText
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.button.MaterialButton

class Registro : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_registro)

        val etCorreo = findViewById<EditText>(R.id.etCorreo)
        val etUsuario = findViewById<EditText>(R.id.etUsuario)
        val etContrasena = findViewById<EditText>(R.id.etContrasenaReg)
        val btnCrearCuenta = findViewById<MaterialButton>(R.id.btnCrearCuenta)
        val btnRegresar = findViewById<MaterialButton>(R.id.btnRegresar)

        // Regresa a MainActivity
        btnRegresar.setOnClickListener {
            finish()
        }

        //Crea cuenta y abrir la actividad Inicio
        btnCrearCuenta.setOnClickListener {
            val correo = etCorreo.text.toString().trim()
            val usuario = etUsuario.text.toString().trim()
            val contrasena = etContrasena.text.toString().trim()

            // Validación
            if (correo.isEmpty() || usuario.isEmpty() || contrasena.isEmpty()) {
                Toast.makeText(this, "Por favor completa todos los campos", Toast.LENGTH_SHORT).show()
            } else {
                // (Aquí iría lógica de guardar los datos en una base de datos, si tuvieras una)

                // Mostrar mensaje y redirigir a Inicio
                Toast.makeText(this, "Registro exitoso. Bienvenido, $usuario", Toast.LENGTH_SHORT).show()

                val intent = Intent(this, Inicio::class.java)
                startActivity(intent)
                finish() // opcional: cerrar Registro para que no vuelva con back
            }
        }
    }
}
